Code/RDGeneral/Dict.h

```diff
@@ -237,6 +238,9 @@ class Dict {
   void clearVal(const std::string &what) {
     for(DataType::iterator it = _data.begin(); it < _data.end() ; ++it) {
       if (it->key == what) {
+        if (_hasNonPodData) {
+          RDValue::cleanup_rdvalue(it->val);
+        }
         _data.erase(it);
         return;
       }
```

Comment:
We need to change setPODVal as well and any place where we are replacing an existing value.

I completely missed resetting an existing nonpod value with a pod value.  I'd even be happy with always calling cleanup_rdvalue in these cases.  The hasNonPodData was really intended for faster destruction.